Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AdapterBuilder.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzFramework/DocumentPropertyEditor/DocumentSchema.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    void AdapterBuilder::Value(Dom::Value value)
+    {
+        CurrentNode().SetNodeValue(value);
+    }
+
+    void AdapterBuilder::Attribute(Name attribute, Dom::Value value)
+    {
+        CurrentNode()[attribute] = AZStd::move(value);
+    }
+
+    bool AdapterBuilder::IsError() const
+    {
+        return !m_error.empty();
+    }
+
+    const AZStd::string& AdapterBuilder::GetError() const
+    {
+        return m_error;
+    }
+
+    Dom::Value&& AdapterBuilder::FinishAndTakeResult()
+    {
+        AZ_Assert(m_entries.empty(), "AdapterBuilder::FinishAndTakeResult called before the builder finished");
+        return AZStd::move(m_value);
+    }
+
+    void AdapterBuilder::Error(AZStd::string_view message)
+    {
+        m_error.append(message);
+        m_error.append("\n");
```

Comment:
Nitpick: I think characters can be appended as well, so `'\n'` which would be just a smidgen faster.

Reply:
Looks like we do support a character overload that [does an append N times for a char](https://github.com/o3de/o3de/blob/6df71ce037f1fcc29bd30e88726b5c085eab789a/Code/Framework/AzCore/AzCore/std/string/string.h#L231) so I'll use that, it's done in a few other places (and in `operator+=`)